pkg/sbom/cyclonedx/unmarshal.go

```diff
@@ -129,9 +129,10 @@ func (b *BOM) parseComponent(c cdx.Component) (*core.Component, error) {
 	}
 
 	// Parse PURL
-	var purl packageurl.PackageURL
+	var purl *packageurl.PackageURL
 	if c.PackageURL != "" {
-		purl, err = packageurl.FromString(c.PackageURL)
+		purl = &packageurl.PackageURL{}
```

Comment:
Looks like we can remove this

Reply:
it seems that no, we have to allocate memory